extension/src/format/sexpression.ts

```diff
@@ -2,94 +2,353 @@ import { closeParenStyle, maximumLineChars, longListFormatStyle, indentSpaces }
+    }
+
+
+    // Finds a child LispContainer if it legitimately touches/contains the Position
+    getExpressionFromPos(loc: Position, parent: boolean = false): LispContainer {
+        let result: LispContainer = null;
+        if (this.contains(loc)) {
+            this.atoms.forEach(lispObj => {
+                if (lispObj.contains(loc) && lispObj instanceof LispContainer) {
+                    result = lispObj.getExpressionFromPos(loc, parent) ?? (parent ? this : lispObj);
+                }
+            });
+        }
+        return result;
+    }
+
+
+    // Finds the parent LispContainer of any existing ILispFragment, typically used with a primary Container like Defun
+    getParentOfExpression(child: ILispFragment): LispContainer {        
+        const pos = new Position(child.line, child.column);
+        return this.getExpressionFromPos(pos, true);
+    }
+
+
+    // Gets a range representing the full LispContainer, especially useful for TextDocument.getText()
+    getRange(): Range {
+        const begin: ILispFragment = this.atoms[0];
+        const close: ILispFragment = this.atoms[this.atoms.length -1];
+        return new Range(begin.line, begin.column, close.line, (close.column + close.symbol.length));
+    }
+
+
+    // Returns true if this LispContainer encapsulates the provided Position
+    contains(position: Position): boolean {
+        return this.getRange().contains(position);
+    }
+
+
+    // This version was necessary to properly alternate over SETQ Name vs Value 
+    private isValidForSetq(atom: ILispFragment): boolean {
+        return !atom.isComment() && !['\'', '(', ')', '.'].includes(atom.symbol) && (atom instanceof LispContainer || atom.symbol.trim().length > 0);
+    }    
+    // This is general purpose utility to make sure primitives such as strings, numbers and decorations are not evaluated
+    private notNumberStringOrProtected(atom: ILispFragment): boolean {        
+        return !atom.isComment() && !['\'', '(', ')', '.'].includes(atom.symbol) && !(/^".*"$/.test(atom.symbol)) && !(/^\'{0,1}\-{0,1}\d+[eE\-]{0,2}\d*$/.test(atom.symbol));
+    }
+
+
+    // This is an iteration helper that makes sure only useful LispAtom/LispContainers's get used for data collection purposes.
+    nextKeyIndex(currentIndex: number, forSetq?: boolean): number {
+        let index = currentIndex + 1;
+        if (index < this.atoms.length) {
+            const atom = this.atoms[index];
+            const flag = forSetq ? this.isValidForSetq(atom) : this.notNumberStringOrProtected(atom);
+            if (atom instanceof LispAtom && flag) {
+                return index;               
+            } else {
+                return this.nextKeyIndex(index, forSetq);
+            }
+        } else {
+            return -1;
+        }
+    }
+
+
+    // Returns a meaningful value from the LispContainer, this is useful for avoiding comments an structural symbols
+    getNthKeyAtom(significantNth: number): ILispFragment {
```

Comment:
based on the introduction of the interface and the return type, i should probably change this to `getNthKeyFragment`